### PR TITLE
fix: refactor CLI options (WIP)

### LINKED ISSUE #26: Refactor the command-line options
The command line configuration is currently

```rust
#[derive(Clap, Debug)]
#[clap(setting = AppSettings::ColoredHelp)]
#[clap(name = "gh-metrics")]
enum Opt {
    /// list all repositories in the given organization and the number of
    /// Pull Requests created in the last 30 days
    List {
        /// name of GitHub organization to analyze
        #[clap(short, long)]
        org: String,

        /// Verbose mode (-v, -vv, -vvv, etc.)
        #[clap(short, long, parse(from_occurrences))]
        verbose: u8,

        #[clap(short, long)]
        google_sheet: Option<String>,
    },
}
```

but I think that the various options (e.g., `org`, `verbose`, `google_sheet`) should really be independent of the command in use. I imagine the command being the way we specify the producer, and the other options being the way we specify the consumer (or options that apply to many producers).

So you'd be able to do things like e.g.

```rust
github-metrics list --csv xxx.csv
github-metrics count-contributors --csv xxx.csv
github-metrics count-contributors --google-sheet xxx
```

### PR DESCRIPTION
**Pull request by chazkiker2**
## Description

cc #26

### CODE REVIEW

**Comment by chazkiker2** [comment]
WIP — I think my GitHub rate is cached for the moment

File: src/main.rs [hunk 1]
```diff
@@ -85,70 +86,70 @@ async fn main() {
...
+    if let Some(sheet_id) = sheet_id {
+        if let Err(e) = ExportToSheets::new(&sheet_id)
+            .consume(&mut rx, column_names.clone().unwrap())
+            .await
+        {
+            println!("Error exporting to sheets: {}", e);
+        }
+    }
+    // if user specified the print flag, they must want to print to terminal
+    if cli.print {
```
Comment (nikomatsakis) [inline src/main.rs:150]:
```suggestion
    else if cli.print {
```

we only do one or the other, right?

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (nikomatsakis) [reply src/main.rs:150]: Actually, maybe we should set things up so that there is an intermediate layer, and you can expert in multiple ways?

**Comment by nikomatsakis** [comment]
I know this is not totally done, @chazkiker2, but I kind of want these changes so I'm going to land it anyway. :)

**Pull request merged**